lib/index.js

```diff
@@ -76,57 +77,269 @@ class AuthPassportLocal {
         return this.configRepository.create({_id: 'secret', configurationValue: this.config.secret});
       })
       .then(() => {
-        strategy = new Strategy(this.context);
-        strategy.load();
-
-        this.context.passwordManager = new PasswordManager(this.config);
+        this.passwordManager = new PasswordManager(this.config);
 
-        return true;
+        return Promise.resolve();
       });
   }
+
+  initStrategies () {
+    this.strategies = {
+      local: {
+        config: {
+          constructor: LocalStrategy,
+          strategyOptions: {},
+          authenticateOptions: {
+            scope: []
+          },
+          fields: ['login', 'password']
+        },
+        methods: {
+          create: 'create',
+          delete: 'delete',
+          exists: 'exists',
+          getById: 'getById',
+          getInfo: 'getInfo',
+          update: 'update',
+          validate: 'validate',
+          verify: 'verify'
+        }
+      }
+    };
+  }
+
   /**
-   * @param {object<string, *>} user
-   * @param {function} callback
+   * @param {KuzzleRequest} request
+   * @param {object} credentials
+   * @param {string} kuid
+   * @param {boolean} isUpdate
+   * @returns {Promise<boolean>}
    */
-  cleanUserCredentials (user, callback) {
-    try {
-      if (user._source !== undefined) {
-        delete user._source.password;
+  validate (request, credentials, kuid, isUpdate) {
```

Comment:
`validate` signature takes 5 parameters: missing `strategy` between `kuid` and `isUpdate`